src/Microsoft.AspNetCore.Http.Abstractions/PathString.cs

```diff
@@ -335,9 +361,28 @@ public override int GetHashCode()
         /// Implicitly calls ToString().
         /// </summary>
         /// <param name="path"></param>
-        public static implicit operator string (PathString path)
+        public static implicit operator string(PathString path)
         {
             return path.ToString();
         }
+
+        private static bool IsLegalPathChar(char c)
+        {
+            return (c >= 0x41 && c <= 0x5A) ||  // A - Z
+                   (c >= 0x61 && c <= 0x7A) ||  // a - z
+                   (c >= 0x30 && c <= 0x39) ||  // 0 - 9
+                   (c >= 0x26 && c <= 0x2C) ||  // ' ( ) * + ,
+                    c == 0x2D ||    // hyphen
+                    c == 0x2E ||    // period
+                    c == 0x5F ||    // underscore
+                    c == 0x7E ||    // tilde
+                    c == 0x40 ||    // @
+                    c == 0x3A ||    // :
+                    c == 0x2F ||    // forward slash
+                    c == 0x21 ||    // !
+                    c == 0x24 ||    // $
+                    c == 0x3B ||    // ;
+                    c == 0x3D;      // =
```

Comment:
I have sometimes seen this implemented as a pre-initialized bool array with the max range (0-126) that you just index into.